File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.py
class Solution:
    def checkSubarraySum(self, nums: List[int], k: int) -> bool:
        


        hashmap = {0:-1}
        total = 0

        
        for i in range(len(nums)):
            total += nums[i]
            remainder = total % k

            if remainder not in hashmap:
                hashmap[remainder] = i
            elif i - hashmap[remainder]  > 1:
                return True
        
        return False


# solution makes hell sense

File: 0523-continuous-subarray-sum/0523-continuous-subarray-sum.cpp
#include <iostream>


#include <map>
using namespace std;


class Solution {
public:
    bool checkSubarraySum(vector<int>& nums, int k) {


        map<int, int> myhashmap;
        
        myhashmap[0] = -1;


        int total = 0;

        for (int i = 0 ; i < nums.size(); i++ ){

            total += nums[i];
            int remainder = total % k;

            if (myhashmap.count(remainder) ==  0){
                myhashmap[remainder] = i;
            }
            else if (i - myhashmap[remainder] > 1)
            {
                return true;
            }
            

        }
        return false;


    }
};
